library/agent/ServiceConfig.ts

```diff
@@ -136,12 +144,31 @@ export class ServiceConfig {
     this.setBlockedIPAddresses(blockedIPAddresses);
   }
 
+  private setMonitoredIPAddresses(monitoredIPAddresses: IPList[]) {
+    this.monitoredIPAddresses = [];
+
+    for (const source of monitoredIPAddresses) {
+      this.monitoredIPAddresses.push({
+        key: source.key,
+        list: new IPMatcher(source.ips),
+      });
+    }
+  }
+
+  updateMonitoredIPAddresses(monitoredIPAddresses: IPList[]) {
+    this.setMonitoredIPAddresses(monitoredIPAddresses);
+  }
+
+  isMonitoredIPAddress(ip: string): boolean {
```

Comment:
Your going through it twice here then?

Reply:
https://github.com/AikidoSec/firewall-node/pull/601/commits/622be4eb909b83f78652fd92d199f6c3e37eb94e